benchmarks/main.ts

```diff
@@ -0,0 +1,324 @@
+/// <reference no-default-lib="true" />
+/// <reference lib="deno.window" />
+
+import { Command } from "https://deno.land/x/cliffy@v0.25.2/command/mod.ts";
+import { string } from "https://deno.land/x/cliffy@v0.25.2/flags/types/string.ts";
+import { sleep } from "https://deno.land/x/sleep@v1.2.1/mod.ts";
+import * as windmill from "https://deno.land/x/windmill@v1.37.0/mod.ts";
+import * as api from "https://deno.land/x/windmill@v1.37.0/windmill-api/index.ts";
+import { InfluxDB, Point, HttpError } from "npm:@influxdata/influxdb-client";
+import { BucketsAPI, OrgsAPI } from "npm:@influxdata/influxdb-client-apis";
+
+async function login(
+  config: api.Configuration,
+  email: string,
+  password: string
+): Promise<string> {
+  return await new windmill.UserApi(config).login({
+    email: email,
+    password: password,
+  });
+}
+
+await new Command()
+  .name("windmill-bench")
+  .description("Run Benchmark to measure throughput of windmill.")
+  .version("v0.0.0")
+  .option("--host <url:string>", "The windmill host to benchmark.", {
+    default: "http://127.0.0.1/",
+  })
+  .option("-j --jobs <jobs:number>", "The number of jobs to run at once.", {
+    default: 1,
+  })
+  .option(
+    "-s --seconds <seconds:number>",
+    "How long to run the benchmark for (in seconds).",
+    {
+      default: 30,
+    }
+  )
+  .option("-e --email <email:string>", "The email to use to login.")
+  .option("-p --password <password:string>", "The password to use to login.")
+  .env(
+    "WM_TOKEN=<token:string>",
+    "The token to use when talking to the API server. Preferred over manual login."
+  )
+  .option(
+    "-t --token <token:string>",
+    "The token to use when talking to the API server. Preferred over manual login."
+  )
+  .env(
+    "WM_WORKSPACE=<workspace:string>",
+    "The workspace to spawn scripts from."
+  )
+  .option(
+    "-w --workspace <workspace:string>",
+    "The workspace to spawn scripts from.",
+    { default: "starter" }
+  )
+  .option("-m --metrics <metrics:string>", "The url to scrape metrics from.", {
+    default: "http://localhost:8001/metrics",
+  })
+  .option("--influx-host <influx_host:string>", "The influx url to write to.")
+  .option("--influx-token <influx_token:string>", "The influx token to use.")
+  .option("--influx-org <influx_org:string>", "The influx org to write to.")
+  .option(
+    "--influx-bucket <influx_bucket:string>",
+    "The influx bucket to write to. Everything in the bucket will be deleted!!"
+  )
```

Comment:
Do we really need influxDB, would it not be sufficient to do everything in-memory and generate a report from it ? I wonder if something like Danfo would do the trick: https://danfo.jsdata.org/getting-started

Reply:
No, influx is way overkill.
danfo looks great, I'll integrate it to replace the export functionality. It's a bit of an awkward implementation anyways & not as powerful as I'd like it to be.

Reply:
Couldn't get danfojs or similar to work in deno. Implemented the few things we need myself.